bpf/bpf_lxc.c

```diff
@@ -125,7 +126,8 @@ get_ct_map4(struct ipv4_ct_tuple *tuple)
 #if defined ENABLE_IPV4 || defined ENABLE_IPV6
 static inline bool redirect_to_proxy(int verdict, int dir)
 {
-	return verdict > 0 && (dir == CT_NEW || dir == CT_ESTABLISHED);
+	return !is_defined(DATAPATH_IPVLAN) && verdict > 0 &&
```

Comment:
I suggest we make this more generic and use `!is_defined(ENABLE_PROXY_REDIRECT)` and define that in `common.h` for `DATAPATH_IPVLAN`. Some of the user of `DATAPLANE_IPVLAN` will overlap with what @aanm's flannel PR will need as well. If we use fine-grained defines and select them with the high level options that will make it easier to merge overall.

Reply:
Yes, fully agree, will fix is up this way.